quickwit-doc-mapper/src/default_doc_mapper/default_mapper.rs

```diff
@@ -137,6 +137,11 @@ impl DefaultDocMapper {
         }
         Ok(())
     }
+
+    /// Default maximum number of partitions.
+    pub fn default_max_num_partitions() -> NonZeroU64 {
```

Comment:
I assume we'll advise users to always go with a small number of partitions to prevent indexing issues down the line. In that case, affirming that Quickwit supports partitioning looks pretty misleading. I'd be more comfortable if we called this feature "bucketing" and ship true partitioning later.

Reply:
Not that Quickwit is comparable to those engines, but for reference, Hive and Spark start choking when the number of partitions is in the thousands, and the size of partitions is reaching < 1 GB.